pkg/virt-handler/virtwrap/manager.go

```diff
@@ -328,21 +327,14 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 	return domName
 }
 
-func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
-	var wantedSpec api.DomainSpec
-	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
-
-	if len(mappingErrs) > 0 {
-		return nil, errors.NewAggregate(mappingErrs)
-	}
-
+func (l *LibvirtDomainManager) SyncVM(vm *v1.VM, domDesign *designer.DomainDesign) (*api.DomainSpec, error) {
```

Comment:
At least for disks we can do the resolution already before it is passed into SyncVM. I want to keep this clean from k8s rest api calls. I also have a feeling that we should resolve secrets also before we call SyncVM. 

The flow would be:

 1) resolve dependencies (volumes, secrets, ...)
 2) validate from the cluster perspective
 3) map to VM and k8s secrets
 4) Pass it  to SyncVM


SyncVM would to the following:

 1) map from VM to Domain; the same for secrets (no k8s cluster connection necessary)
 3) validation from the libvirt perspective
 4) synchronize with libvirt (in the future also hotplug, ...)

Steps  1-3 can also be done in another struct/object, and not entirely in the controller itself, if it helps the clarity. I think SyncVM is already complex enough by dealing with libvirt. It also makes testing the individual components and LibvirtManager easier, if we separate that.